service/gcs/core/gcs/gcs.go

```diff
@@ -472,31 +492,39 @@ func (c *gcsCore) ModifySettings(id string, request prot.ResourceModificationReq
 		return errors.WithStack(gcserr.NewContainerDoesNotExistError(id))
 	}
 
+	settings, ok := request.Settings.(prot.ResourceModificationSettings)
```

Comment:
You are actually changing the behavior of the `default` case here but I think its fine. You will just see a different error first.